File: 卡码/代码随想录/1/06-二叉树/Python/01-二叉树遍历方式/LeetCode_0107.py
#107. 二叉树的层序遍历 II
from typing import *
from ..TreeNode import *

class Solution:
  def levelOrderBottom(self, root: Optional[TreeNode]) -> List[List[int]]:
    res=[]
    queue=[]
    if root!=None:
      queue.append(root)
    while len(queue)>0:
      size=len(queue)
      v=[]
      while size>0:
        size-=1
        node=queue[0]
        queue=queue[1:]
        v.append(node.val)
        if node.left!=None:
          queue.append(node.left)
        if node.right!=None:
          queue.append(node.right)
      res.insert(0,v)
    return res

File: 卡码/代码随想录/C++/二叉树/01-二叉树遍历方式/LeetCode_0107.cpp
//107. ¶ş²æÊ÷µÄ²ãĞò±éÀú II
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
  vector<vector<int>> levelOrderBottom(TreeNode* root) {
		queue<TreeNode*> q;
		vector<vector<int>> res;
		vector<int> v;
		if(root) q.push(root);
		int size;
		while(!q.empty()){
			size=q.size();
			v.clear();
			while(size-->0){
				TreeNode *node=q.front();
				q.pop();
				v.push_back(node->val);
				if(node->left) q.push(node->left);
				if(node->right) q.push(node->right);
			}
			res.push_back(v);
		}
		reverse(res.begin(),res.end());
		return res;
  }
};

int main(){
  system("pause");
  return 0;
}
